client/components/data/query-plans/index.jsx

```diff
@@ -1,36 +1,20 @@
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 import { requestPlans } from 'calypso/state/plans/actions';
 import { isRequestingPlans } from 'calypso/state/plans/selectors';
 
-class QueryPlans extends Component {
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillMount() {
-		if ( ! this.props.requestingPlans ) {
-			this.props.requestPlans();
-		}
+const request = () => ( dispatch, getState ) => {
+	if ( ! isRequestingPlans( getState() ) ) {
```

Comment:
This `isRequesting` check is not needed, the `requestPlans` action is handled by data layer which deduplicates requests automatically if an identical one is already in flight.

Reply:
I kept it because it existed in the previous version, but I don't see why we can't clean it up if it's unnecessary.

Removed in 6f5247a6bdc292ef243f04b4e5cc74fd2f79d10b